src/app/rpmostree-builtin-rebase.cxx

```diff
@@ -118,44 +118,44 @@ rpmostree_builtin_rebase (int             argc,
       if (argc == 3)
         revision = argv[2];
 
-      /* Canonicalize */
-      new_provided_refspec = new_refspec_owned =
-        rpmostree_refspec_canonicalize (new_provided_refspec, error);
-      if (!new_provided_refspec)
-        return FALSE;
+      new_provided_refspec = new_refspec_owned = g_strdup (new_provided_refspec);
     }
   else
     {
       if (opt_remote)
         {
           new_provided_refspec = new_refspec_owned =
-            g_strconcat (RPMOSTREE_REFSPEC_OSTREE_PREFIX, opt_remote, ":", opt_branch ?: "", NULL);
+            g_strconcat (opt_remote, ":", opt_branch ?: "", NULL);
         }
       else
         new_provided_refspec = opt_branch;
     }
   (void)new_refspec_owned; /* Pacify static analysis */
 
-  const char *remainder = NULL;
   RpmOstreeRefspecType refspectype;
-  if (!rpmostree_refspec_classify (new_provided_refspec, &refspectype, &remainder, error))
+  if (!rpmostree_refspec_classify (new_provided_refspec, &refspectype, error))
     return FALSE;
 
-  /* catch "ostree://"; we'd error out much later in the daemon otherwise */
-  if (strlen (remainder) == 0)
+  /* catch empty refspec now; we'd error out much later in the daemon otherwise */
+  if (strlen (new_provided_refspec) == 0)
     return glnx_throw (error, "Refspec is empty");
 
+  /* When using the container refspec type, if rebasing to a specific commit, we expect a
+   * specific digest tag in the refspec, not in a separate argument */
+  if (revision && refspectype == RPMOSTREE_REFSPEC_TYPE_CONTAINER)
+    return glnx_throw (error, "Unexpected ostree revision alongside container refspec type");
```

Comment:
Totally fine as is, but if this isn't expected to be reachable via user input, it could also be `g_error()` i.e. we abort.